modules/age.nix

```
@@ -201,6 +212,41 @@ in
         "agenixChownKeys"
       ];
     };
+
+    # services that watch for file changes and exectue the configured action
+    systemd.services = lib.mkMerge
+      (lib.mapAttrsToList
+        (name: {action, service, file, path, mode, owner, group, ...}:
+          let
+            fileHash = builtins.hashString "sha256" (builtins.readFile file);
+            restartTriggers = [ fileHash path mode owner group ];
+          in
+            lib.mkMerge [
+              (lib.mkIf (service != "") {
+                ${service} = { inherit restartTriggers; };
+              })
+              (lib.mkIf (action != "") {
+                "agenix-${name}-action" = {
+                  inherit restartTriggers;
+
+                  # We execute the action on reload so that it doesn't happen at
+                  # startup. The only disadvantage is that it won't trigger the
+                  # first time the service is created.
+                  reload = action;
+                  reloadIfChanged = true;
+
+                  serviceConfig = {
+                    Type = "oneshot";
+                    RemainAfterExit = true;
+                  };
+
+                  script = " "; # it complains if we only set ExecReload
```

Minor nit, but I think it may be better to be clearer here? Not sure, thoughts welcome on if this is necessary.

```suggestion
                  script = " "; # systemd complains if we only set ExecReload
```